calyx-py/calyx/builder.py

```diff
@@ -536,6 +536,18 @@ def incr(self, reg, val=1, signed=False, cellname=None):
             incr_group.done = reg.done
         return incr_group
 
+    def incr_static(self, reg, val=1, signed=False, cellname=None):
+        """Inserts wiring into `self` to perform `reg := reg + val` statically."""
+        cellname = cellname or f"{reg.name}_incr"
+        width = reg.infer_width_reg()
+        add_cell = self.add(width, cellname, signed)
+        with self.static_group(f"{cellname}_group", 1) as incr_group:
+            add_cell.left = reg.out
+            add_cell.right = const(width, val)
+            reg.write_en = 1
+            reg.in_ = add_cell.out
```

Comment:
I would have been nice to have reused the existing `insert_incr` with a `static` flag as an argument. I got a good way through it, but then the issue is that the dynamic version needs a line `incr_group.done = ...` but the static version does not. 

I'd usually do that with the Python idiom of `X if b else Y`, but that doesn't tango super well with the eDSL's own `with foo as bar: baz` idiom

Reply:
Just paging @sampsyo in case he knows a neat way! Like basically is there some `Y` such that I could do:
```
with foo as bar:
  ...
  ...
  X if not static else Y
```

and the `Y` would be ignored when generating Calyx code?

If I had a nice way of doing this, I could go through and do it for lots of helpers in the eDSL!

Reply:
Thank you for the sync chat; it worked like a charm!